python/cudf/cudf/_lib/sort.pyx

```diff
@@ -145,6 +146,53 @@ def order_by(list columns_from_table, object ascending, str na_position):
     return Column.from_unique_ptr(move(c_result))
 
 
+@acquire_spill_lock()
+def sort(
```

Comment:
I was shocked we didn't have Cython bindings for `cudf::sort` yet. I suspect there are lots of places in cudf where we need to replace an "argsort and gather" with a plain table sort. That would eliminate a layer of Python / kernel overhead.

Reply:
I imagine that there are also a bunch of cases where `argsort + gather` is actually a `sort_by_key` in disguise.